solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -255,9 +257,13 @@ protected HttpSolrCall getHttpSolrCall(HttpServletRequest request, HttpServletRe
       throw new SolrException(ErrorCode.SERVER_ERROR, "Core Container Unavailable");
     }
     if (isV2Enabled && (path.startsWith("/____v2/") || path.equals("/____v2"))) {
-      return new V2HttpCall(this, cores, request, response, false);
+      return isCoordinator ?
+              new CoordinatorHttpSolrCall(this, cores, request, response, false) :
+              new V2HttpCall(this, cores, request, response, false);
     } else {
-      return new HttpSolrCall(this, cores, request, response, retry);
+      return isCoordinator ?
+              new CoordinatorHttpSolrCall(this, cores, request, response, retry) :
+              new HttpSolrCall(this, cores, request, response, retry);
```

Comment:
This feels very fragile plugging into SDF this way. I thought part of the goal with node roles was that we make things more modular instead of coupled.